PPC64le SegFaults

### To reproduce

1. login to mr-0xp1
2. cd `/home/jenkins/slave_dir_from_mr-0xc1/workspace/h2o-py-data-table_rel-0.3.2-HSRX7AFTBJEESK7VGQBWNRPFHJGGB6NZY3F2GPECNPOQ2U7CP6VA`
3. Run `docker run --ulimit core=-1 -it --rm -u $(id -u):$(id -g) -v $(pwd):$(pwd) -w $(pwd) docker.h2o.ai/opsh2oai/datatable-build-ppc64le_centos7`
4. Build dtbl
    ```
    cd build
    . activate datatable-py36
    make debug
    ```
5. Run tests
    ```
    cd ../test-ppc64le_linux-datatable-py36
    . activate datatable-py36
    make test MODULE=datatable
    ```
6. Enjoy seg fault (it is not deterministic)
    ```
    tests/munging/test_dt_rows.py .................................................................make: *** [test] Segmentation fault (core dumped)
    ```

### 1. SegFault

From gdb:

```
(gdb) info frame
Stack level 0, frame at 0x3fffa004e070:
 pc = 0x3ffed917002c; saved pc 0x3fffa4ba4838
 called by frame at 0x3fffa004e1b0
 Arglist at 0x3fffa004e070, args:
 Locals at 0x3fffa004e070, Previous frame's sp is 0x3fffa004e070
(gdb)
```

The mapping shows `datatable.so` at `saved PC`:
```
 0x3fffa4b00000     0x3fffa4cd0000   0x1d0000        0x0 /opt/h2oai/dai/python/envs/datatable-py36/lib/python3.6/site-packages/datatable/lib/_datatable.cpython-36m-powerpc64le-linux-gnu.so
```
---

### 2. SegFault

If tests fail on:
```
tests/test_dt.py .............................                                                                                                                                                                                                                           [  2%]
tests/test_dt_options.py .........                                                                                                                                                                                                                                       [  3%]
tests/test_dt.py ........................s....                                                                                                                                                                                                                           [  6%]
tests/test_dt_create.py ............................................                                                                                                                                                                                                     [ 11%]
tests/test_dt_expr.py ........................make: *** [test] Segmentation fault (core dumped)
```

GDB reports problems in static cast (section which is parallelized by OMP):

```
#1 0x00003fffa9b9d648 in .omp_outlined..38(void) (.global_tid.=0x3fffd3e30378, .bound_tid.=0x3fffd3e30380, this=0x3fffa9bb9000 <RowIndex::inverse(long) const+560>, xi=@0x3fffd3e30388: 0x3fffa51d5308, xo=@0x119237f30: 0x63, una=@0x118ffb070: 4054029733469093921,
    umin=@0x3fffa76951a0: 176767840160579700) at c/sort.cc:399
399	        xo[j] = t == una? 0 : static_cast<TO>(t - umin + 1);
```

--- a/Makefile
+++ b/Makefile
@@ -220,7 +220,7 @@ ubuntu_docker_publish: Dockerfile-ubuntu.$(PLATFORM).tag
 	docker push $(CONTAINER_NAME_TAG)
 
 ARCH_NAME ?= $(shell uname -m)
-DOCKER_IMAGE_TAG ?= 0.6.0-PR-1010.3
+DOCKER_IMAGE_TAG ?= 0.6.0-PR-1010.4
 CENTOS_DOCKER_IMAGE_NAME ?= docker.h2o.ai/opsh2oai/datatable-build-$(ARCH_NAME)_centos7:$(DOCKER_IMAGE_TAG)
 UBUNTU_DOCKER_IMAGE_NAME ?= docker.h2o.ai/opsh2oai/datatable-build-$(ARCH_NAME)_ubuntu:$(DOCKER_IMAGE_TAG)
 

--- a/Makefile
+++ b/Makefile
@@ -220,7 +220,7 @@ ubuntu_docker_publish: Dockerfile-ubuntu.$(PLATFORM).tag
 	docker push $(CONTAINER_NAME_TAG)
 
 ARCH_NAME ?= $(shell uname -m)
-DOCKER_IMAGE_TAG ?= 0.6.0-PR-1147.4
+DOCKER_IMAGE_TAG ?= 0.6.0-PR-1010.3
 CENTOS_DOCKER_IMAGE_NAME ?= docker.h2o.ai/opsh2oai/datatable-build-$(ARCH_NAME)_centos7:$(DOCKER_IMAGE_TAG)
 UBUNTU_DOCKER_IMAGE_NAME ?= docker.h2o.ai/opsh2oai/datatable-build-$(ARCH_NAME)_ubuntu:$(DOCKER_IMAGE_TAG)
 

--- a/ci/Dockerfile-centos7.in
+++ b/ci/Dockerfile-centos7.in
@@ -13,18 +13,19 @@ RUN \
         bzip2 && \
     wget -q https://repo.continuum.io/miniconda/Miniconda3-4.3.27-Linux-ARCH_SUBST.sh && \
     bash Miniconda3-4.3.27-Linux-ARCH_SUBST.sh -b -p /opt/h2oai/dai/python && \
-    wget -q https://s3.amazonaws.com/artifacts.h2o.ai/releases/ai/h2o/dai-thirdparty-deps-llvm/1.0-master-21/ARCH_SUBST-centos7/llvm.tar.bz2 && \
+    wget -q https://s3.amazonaws.com/artifacts.h2o.ai/releases/ai/h2o/dai-thirdparty-deps-llvm/1.0-PR-2-4/ARCH_SUBST-centos7/llvm.tar.bz2 && \
     tar xvf llvm.tar.bz2 && \
     cp -r llvm/* /opt/h2oai/dai/ && \
     rm -rf llvm*
 
-ENV LLVM4=/opt/h2oai/dai
+ENV LLVM6=/opt/h2oai/dai
+ENV LLVM_CONFIG=$LLVM6/bin/llvm-config
 ENV PATH=/opt/h2oai/dai/python/bin:$PATH
 ENV PATH=/usr/local/bin:$PATH
-ENV PATH=$LLVM4/bin:$PATH
+ENV PATH=$LLVM6/bin:$PATH
 ENV PANDAS_VERSION=0.20.0
 
-RUN pip install llvmlite==0.20.0 wheel virtualenv && \
+RUN pip install llvmlite==0.23.0 wheel virtualenv && \
     \
     conda create -n datatable-py35-with-pandas python=3.5 && \
     . activate datatable-py35-with-pandas && \

--- a/ci/Dockerfile-centos7.in
+++ b/ci/Dockerfile-centos7.in
@@ -13,7 +13,7 @@ RUN \
         bzip2 && \
     wget -q https://repo.continuum.io/miniconda/Miniconda3-4.3.27-Linux-ARCH_SUBST.sh && \
     bash Miniconda3-4.3.27-Linux-ARCH_SUBST.sh -b -p /opt/h2oai/dai/python && \
-    wget -q https://s3.amazonaws.com/artifacts.h2o.ai/releases/ai/h2o/dai-thirdparty-deps/1.0-PR-2-9/ARCH_SUBST-centos7/llvm.tar.bz2 && \
+    wget -q https://s3.amazonaws.com/artifacts.h2o.ai/releases/ai/h2o/dai-thirdparty-deps/1.0-master-2/ARCH_SUBST-centos7/llvm.tar.bz2 && \
     tar xvf llvm.tar.bz2 && \
     cp -r llvm/* /opt/h2oai/dai/ && \
     rm -rf llvm*

--- a/ci/Dockerfile-centos7.in
+++ b/ci/Dockerfile-centos7.in
@@ -13,7 +13,7 @@ RUN \
         bzip2 && \
     wget -q https://repo.continuum.io/miniconda/Miniconda3-4.3.27-Linux-ARCH_SUBST.sh && \
     bash Miniconda3-4.3.27-Linux-ARCH_SUBST.sh -b -p /opt/h2oai/dai/python && \
-    wget -q https://s3.amazonaws.com/artifacts.h2o.ai/releases/ai/h2o/dai-thirdparty-deps-llvm/1.0-PR-2-4/ARCH_SUBST-centos7/llvm.tar.bz2 && \
+    wget -q https://s3.amazonaws.com/artifacts.h2o.ai/releases/ai/h2o/dai-thirdparty-deps-llvm/1.0-PR-2-9/ARCH_SUBST-centos7/llvm.tar.bz2 && \
     tar xvf llvm.tar.bz2 && \
     cp -r llvm/* /opt/h2oai/dai/ && \
     rm -rf llvm*

--- a/ci/Dockerfile-centos7.in
+++ b/ci/Dockerfile-centos7.in
@@ -13,7 +13,7 @@ RUN \
         bzip2 && \
     wget -q https://repo.continuum.io/miniconda/Miniconda3-4.3.27-Linux-ARCH_SUBST.sh && \
     bash Miniconda3-4.3.27-Linux-ARCH_SUBST.sh -b -p /opt/h2oai/dai/python && \
-    wget -q https://s3.amazonaws.com/artifacts.h2o.ai/releases/ai/h2o/dai-thirdparty-deps-llvm/1.0-PR-2-9/ARCH_SUBST-centos7/llvm.tar.bz2 && \
+    wget -q https://s3.amazonaws.com/artifacts.h2o.ai/releases/ai/h2o/dai-thirdparty-deps/1.0-PR-2-9/ARCH_SUBST-centos7/llvm.tar.bz2 && \
     tar xvf llvm.tar.bz2 && \
     cp -r llvm/* /opt/h2oai/dai/ && \
     rm -rf llvm*

--- a/ci/Dockerfile-ubuntu.in
+++ b/ci/Dockerfile-ubuntu.in
@@ -8,6 +8,7 @@ MAINTAINER H2O.ai <ops@h2o.ai>
 
 ENV DEBIAN_FRONTEND noninteractive
 ENV LD_LIBRARY_PATH=/usr/local/cuda/lib64:$LD_LIBRARY_PATH
+ENV LLVM_VERSION=6.0.1
 
 RUN \
   apt-get -q -y update && \
@@ -68,15 +69,15 @@ RUN \
 
 # Install LLVM for pydatatable
 RUN \
-  wget http://releases.llvm.org/4.0.0/clang+llvm-4.0.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz && \
-  tar xf clang+llvm-4.0.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz && \
-  rm clang+llvm-4.0.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz && \
-  cp /clang+llvm-4.0.0-x86_64-linux-gnu-ubuntu-16.04/lib/libomp.so /usr/lib
+  wget http://releases.llvm.org/${LLVM_VERSION}/clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-16.04.tar.xz && \
+  tar xf clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-16.04.tar.xz && \
+  rm clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-16.04.tar.xz && \
+  cp /clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-16.04/lib/libomp.so /usr/lib
 
 ENV \
-  LLVM4=/clang+llvm-4.0.0-x86_64-linux-gnu-ubuntu-16.04 \
-  LLVM_CONFIG=/clang+llvm-4.0.0-x86_64-linux-gnu-ubuntu-16.04/bin/llvm-config \
-  CLANG=/clang+llvm-4.0.0-x86_64-linux-gnu-ubuntu-16.04/bin/clang
+  LLVM6=/clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-16.04 \
+  LLVM_CONFIG=/clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-16.04/bin/llvm-config \
+  CLANG=/clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-16.04/bin/clang
 
 # A separated layer to collect core dumps
 RUN mkdir -p /tmp/cores && chmod a+rwx /tmp/cores

--- a/ci/Jenkinsfile2
+++ b/ci/Jenkinsfile2
@@ -39,7 +39,7 @@ X86_64_CENTOS_DOCKER_IMAGE = "docker.h2o.ai/opsh2oai/datatable-build-x86_64_cent
 EXPECTED_SHAS = [
         files : [
                 'ci/Dockerfile-ubuntu.in' : 'd5b19e748c079d65964e6c1e458f525854aee0f3',
-                'ci/Dockerfile-centos7.in': '834004b0e1407081b64a9cf490231818e2110de1',
+                'ci/Dockerfile-centos7.in': '960aa91010a372c674bb8362dbb91a2175c88f0d',
         ]
 ]
 

--- a/ci/Jenkinsfile2
+++ b/ci/Jenkinsfile2
@@ -34,7 +34,7 @@ RELEASE_BRANCH_PREFIX = 'rel-'
 CREDS_ID = 'h2o-ops-personal-auth-token'
 GITCONFIG_CRED_ID = 'master-gitconfig'
 RSA_CRED_ID = 'master-id-rsa'
-DOCKER_IMAGE_TAG = '0.6.0-PR-1147.4'
+DOCKER_IMAGE_TAG = '0.6.0-PR-1010.3'
 X86_64_CENTOS_DOCKER_IMAGE = "docker.h2o.ai/opsh2oai/datatable-build-x86_64_centos7:${DOCKER_IMAGE_TAG}"
 EXPECTED_SHAS = [
         files : [

--- a/ci/Jenkinsfile2
+++ b/ci/Jenkinsfile2
@@ -39,7 +39,7 @@ X86_64_CENTOS_DOCKER_IMAGE = "docker.h2o.ai/opsh2oai/datatable-build-x86_64_cent
 EXPECTED_SHAS = [
         files : [
                 'ci/Dockerfile-ubuntu.in' : 'd5b19e748c079d65964e6c1e458f525854aee0f3',
-                'ci/Dockerfile-centos7.in': '960aa91010a372c674bb8362dbb91a2175c88f0d',
+                'ci/Dockerfile-centos7.in': 'd9ef4a7f981719e05fff49487c6176b7ba445817',
         ]
 ]
 

--- a/ci/Jenkinsfile2
+++ b/ci/Jenkinsfile2
@@ -38,8 +38,8 @@ DOCKER_IMAGE_TAG = '0.6.0-PR-1147.4'
 X86_64_CENTOS_DOCKER_IMAGE = "docker.h2o.ai/opsh2oai/datatable-build-x86_64_centos7:${DOCKER_IMAGE_TAG}"
 EXPECTED_SHAS = [
         files : [
-                'ci/Dockerfile-ubuntu.in' : '6446a6d1dd63c14db76d273c00f861817a4464ba',
-                'ci/Dockerfile-centos7.in': 'c6bf8d1ac424f81f3742b96218c052ef3f1c834c',
+                'ci/Dockerfile-ubuntu.in' : 'd5b19e748c079d65964e6c1e458f525854aee0f3',
+                'ci/Dockerfile-centos7.in': '834004b0e1407081b64a9cf490231818e2110de1',
         ]
 ]
 

--- a/ci/Jenkinsfile2
+++ b/ci/Jenkinsfile2
@@ -34,7 +34,7 @@ RELEASE_BRANCH_PREFIX = 'rel-'
 CREDS_ID = 'h2o-ops-personal-auth-token'
 GITCONFIG_CRED_ID = 'master-gitconfig'
 RSA_CRED_ID = 'master-id-rsa'
-DOCKER_IMAGE_TAG = '0.6.0-PR-1010.3'
+DOCKER_IMAGE_TAG = '0.6.0-PR-1010.4'
 X86_64_CENTOS_DOCKER_IMAGE = "docker.h2o.ai/opsh2oai/datatable-build-x86_64_centos7:${DOCKER_IMAGE_TAG}"
 EXPECTED_SHAS = [
         files : [

--- a/ci/Jenkinsfile2
+++ b/ci/Jenkinsfile2
@@ -79,7 +79,7 @@ properties([
     parameters([
         booleanParam(name: 'FORCE_BUILD_PPC64LE', defaultValue: false, description: '[BUILD] Trigger build of PPC64le artifacts'),
         booleanParam(name: 'DISABLE_ALL_TESTS', defaultValue: false, description: '[BUILD] Disable all tests'),
-        booleanParam(name: 'DISABLE_PPC64LE_TESTS', defaultValue: true, description: '[BUILD] Disable PPC64LE tests'),
+        booleanParam(name: 'DISABLE_PPC64LE_TESTS', defaultValue: false, description: '[BUILD] Disable PPC64LE tests'),
         booleanParam(name: 'DISABLE_COVERAGE', defaultValue: false, description: '[BUILD] Disable coverage'),
         booleanParam(name: 'FORCE_ALL_TESTS_IN_PR', defaultValue: false, description: '[BUILD] Trigger all tests even for PR')
     ]),

